cleanlab/object_detection/summary.py

```diff
@@ -34,6 +43,412 @@
+        Predictions output by a trained object detection model.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    auxiliary_inputs: optional
+        Auxiliary inputs to be used in the computation of counts.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    class_names: optional
+        A dictionary mapping one-hot-encoded class labels back to their original class names in the format ``{"integer-label": "original-class-name"}``.
+
+    Returns
+    -------
+    bbox_sizes: Tuple[Dict[Any, List], Dict[Any, List]]
+        A tuple containing two dictionaries. Each maps each class label to a list of the sizes of annotated bounding boxes for that class in the label and prediction datasets, respectively.
+    """
+    if auxiliary_inputs is None:
+        auxiliary_inputs = _get_valid_inputs_for_compute_scores(ALPHA, labels, predictions)
+
+    labl_area: DefaultDict[Any, list] = collections.defaultdict(list)
+    pred_area: DefaultDict[Any, list] = collections.defaultdict(list)
+    for sample in auxiliary_inputs:
+        _get_bbox_areas(sample["lab_labels"], sample["lab_bboxes"], labl_area, class_names)
+        _get_bbox_areas(sample["pred_labels"], sample["pred_bboxes"], pred_area, class_names)
+
+    return labl_area, pred_area
+
+
+def class_label_distribution(
+    labels=None,
+    predictions=None,
+    *,
+    auxiliary_inputs=None,
+    class_names: Optional[Dict[Any, Any]] = None,
+) -> Tuple[Dict[Any, float], Dict[Any, float]]:
+    """Returns the distribution of class labels associated with all annotated bounding boxes (or predicted bounding boxes) in the dataset.
+
+    This method can help you understand which classes are: rare or over/under-predicted by the model overall.
+
+    Parameters
+    ----------
+    labels:
+        Annotated boxes and class labels in the original dataset, which may contain some errors.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    predictions:
+        Predictions output by a trained object detection model.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    auxiliary_inputs: optional
+        Auxiliary inputs to be used in the computation of counts.
+        Refer to documentation for this argument in :py:func:`object_counts_per_image <cleanlab.object_detection.summary.object_counts_per_image>` for further details.
+
+    class_names: optional
+        Optional dictionary mapping one-hot-encoded class labels back to their original class names in the format ``{"integer-label": "original-class-name"}``.
```

Comment:
Is "integer-label" a string or int? If it should be an int I suggest writing ``{integer-label: "original-class-name"}`` to avoid confusion

Reply:
It is a string right now. I followed the format of the `class_names` argument in the `visualize()` method.